src/main/java/com/bullhornsdk/data/api/helper/RestApiSession.java

```diff
@@ -128,19 +136,19 @@ private void createSession() {
 				login();
 				break;
 			} catch (Exception e) {
-
 				if (tryNumber < SESSION_RETRY) {
 					log.error("Error creating REST session. Try number: " + tryNumber + " out of " + SESSION_RETRY + " trying again.", e);
 				} else {
 					log.error("Final error creating REST session. Shutting down.", e);
+
 					throw new RestApiException("Failed to create rest session", e);
 				}
 			}
 		}
 	}
 
 	private String getAuthorizationCode() throws RestApiException {
-		String authorizeUrl = restCredentials.getRestAuthorizeUrl();
+		String authorizeUrl = getRestAuthorizeUrl();
```

Comment:
Instead of making this always lookup the RestAuthorizeUrl, can we instead make it lookup only if the `restCredentials.getRestAuthorizeUrl()` isn't provided by the user? That way we don't have to make additional Rest calls during initialization if we don't need to.

Reply:
Nevermind, just realized that's what's happening now!